src/openms/source/KERNEL/MSExperiment.cpp

```diff
@@ -672,12 +672,22 @@ namespace OpenMS
     spectra_ = spectra;
   }
 
+  void MSExperiment::setSpectra(std::vector<MSSpectrum> && spectra)
+  {
+    spectra_ = std::move(spectra);
+  }
+
   /// adds a spectrum to the list
   void MSExperiment::addSpectrum(const MSSpectrum & spectrum)
   {
     spectra_.push_back(spectrum);
   }
 
+  void MSExperiment::addSpectrum(MSSpectrum && spectrum)
+  {
+    spectra_.emplace_back(std::move(spectrum));
```

Comment:
nitpicking (not important):
one convention (which I find reasonable), is that when moving complete objects, use `.push_back()`.
Only if constructing in place, use `emplace_back`.